doc: Upgrade from an old Gitea (#16918)
go-gitea/gitea: docs/content/doc/upgrade/from-gitea.en-us.md

---
date: "2021-09-02T16:00:00+08:00"
title: "Upgrade from an old Gitea"
slug: "upgrade-from-gitea"
weight: 10
toc: false
draft: false
menu:
  sidebar:
    parent: "upgrade"
    name: "From Gitea"
    weight: 10
    identifier: "upgrade-from-gitea"
---

# Upgrade from an old Gitea

**Table of Contents**

{{< toc >}}

To update Gitea, download a newer version, stop the old one, perform a backup, and run the new one.
Every time a Gitea instance starts up, it checks whether a database migration should be run. 
If a database migration is required, Gitea will take some time to complete the upgrade and then serve.

## Backup for downgrade

Gitea keeps compatibility for patch versions whose first two fields are the same (`a.b.x` -> `a.b.y`), 
these patch versions can be upgraded and downgraded with the same database structure. 
Otherwise (`a.b.?` -> `a.c.?`), a newer Gitea version will upgrade the old database 
to a new structure that may differ from the old version.

For example:

| From | To | Result |
| --- | --- | --- |
| 1.4.0 | 1.4.1 | ✅ |

> | 1.4.1 | 1.4.0 | ⚠️ Not recommended, take your own risk! Although it may work if the database structure doesn't change, it's highly recommended to use a backup to downgrade. |

Reviewer: ```suggestion
| 1.4.1 | 1.4.0 | ⚠️ Not recommended, take your own risk! Although it should work since the database structure doesn't change. |
```
